File: 0240 Search a 2D Matrix II.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if (matrix == null || matrix.length == 0 || matrix[0].length == 0) return false;
        int m = matrix.length;
        int n = matrix[0].length;
        int i = 0, j = n - 1;
        
        while (i < m && j >= 0) {
            if (target == matrix[i][j]) return true;
            else if (target > matrix[i][j]) ++i;
            else --j;
        }
        return false;
    }
}

File: 0240 Search a 2D Matrix II.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix or len(matrix) == 0 or len(matrix[0]) == 0: return False
        m, n = len(matrix), len(matrix[0])
        i, j = 0, n - 1
        
        while i < m and j >= 0:
            if target == matrix[i][j]: return True
            elif target > matrix[i][j]: i += 1
            else: j -= 1
        
        return False